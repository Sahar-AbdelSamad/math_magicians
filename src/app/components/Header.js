import React from 'react';
import { Link } from 'react-router-dom';

const Header = () => (
  <header>
    <h1>Math Magicians</h1>
    <nav>
      <Link to="/">Home</Link>
      <Link to="/calculator">Calculator</Link>
      <Link to="/quote">About</Link>
    </nav>
  </header>
);

export default Header;
