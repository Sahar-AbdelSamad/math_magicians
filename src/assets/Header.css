.header {
  display: flex;
  align-items: center;
  font-size: 1.5rem;
  margin-bottom: 6%;
  border-radius: 2%;
  color: #5c1851;
}

.navigation {
  display: flex;
  gap: 10%;
  margin-left: auto;
  margin-right: 10%;
}

a {
  text-decoration: none;
  color: #5c1851;
}

.navigation a:not(:last-of-type) {
  padding-right: 10%;
  border-right: 1px solid #361500;
}
