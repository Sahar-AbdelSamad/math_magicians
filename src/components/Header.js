import { Link } from 'react-router-dom';
import '../assets/Header.css';

const Header = () => (
  <header className="header">
    <Link to="/math_magicians"><h1>Math Magicians</h1></Link>
    <nav className="navigation">
      <Link to="/math_magicians">Home</Link>
      <Link to="/calculator">Calculator</Link>
      <Link to="/quote">Quote</Link>
    </nav>
  </header>
);

export default Header;
