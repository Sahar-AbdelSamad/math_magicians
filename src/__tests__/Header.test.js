import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import Header from '../app/components/Header';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <Router>
        <Header />
      </Router>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
