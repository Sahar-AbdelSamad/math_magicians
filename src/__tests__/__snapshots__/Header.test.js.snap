// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<header
  className="header"
>
  <h1>
    Math Magicians
  </h1>
  <nav
    className="navigation"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      Home
    </a>
    <a
      href="/calculator"
      onClick={[Function]}
    >
      Calculator
    </a>
    <a
      href="/quote"
      onClick={[Function]}
    >
      Quote
    </a>
  </nav>
</header>
`;
